File: shared/base/src/androidMain/kotlin/kmp/shared/base/domain/system/ConfigImpl.kt
package kmp.shared.base.domain.system

import kmp.shared.base.BuildConfig

internal class ConfigImpl : Config {
    override val isRelease: Boolean
        get() = BuildConfig.BUILD_TYPE == "release"

    override val apiVariant: ApiVariant
        get() = ApiVariant.entries.firstOrNull { it.name.lowercase() == BuildConfig.FLAVOR }
            ?: ApiVariant.Alpha
}

File: ios/DomainLayer/SharedDomain/Sources/SharedDomain/ConfigImpl.swift
import KMPShared
import Utilities

public class ConfigImpl: Config {
    
    public init() {}

    public var apiVariant: KMPShared.ApiVariant {
        switch Environment.api {
        case .alpha: KMPShared.ApiVariant.alpha
        case .production: KMPShared.ApiVariant.production
        }
    }
    
    public var isRelease: Bool {
        Environment.build == .release
    }
}
